api/src/pcapi/local_providers/titelive_things/titelive_things.py

```diff
@@ -95,6 +96,9 @@ def __init__(self):  # type: ignore [no-untyped-def]
         self.data_lines = None
         self.products_file = None
         self.product_extra_data = offers_models.OfferExtraData()
+        self.product_whitelist_eans = [
```

Comment:
**Suggestion++ :**
Il me semble que @Meewan avait parlé d'une recherche avec _in_ plus rapide avec un `set` qu'avec une `list`.
Ça ne coûte pas grand chose de changer `[...]` en `{...}` si c'est confirmé.

Reply:
C'est le "si c'est confirmé" qui fait que je sais pas trop : https://stackoverflow.com/questions/2831212/python-sets-vs-lists

Dans le doute, j'attends qu'un expert passe par la :)

Reply:
je confirme que dans ton cas ça sera plus rapide avec un set. la création du set sera plus cher mais LES verifications seront bien plus rapides

Reply:
En parcourant rapidement le lien que tu partages, il semble y avoir un consensus sur « Sets are significantly faster when it comes to determining if an object is present in the set (as in x in s) », ce qui est exactement l'utilisation que tu en fais, puisque tu n'itères pas dessus mais ne fais que `book_unique_identifier in self.product_whitelist_eans`.
Je laisse @Meewan assumer sa suggestion récente :)

Reply:
J'ai commit un changement pour `set`, merci !